bson/src/main/scala/org/mongodb/scala/bson/codecs/macrocodecs/CaseClassCodec.scala

```diff
@@ -71,18 +71,18 @@ private[codecs] object CaseClassCodec {
 
     // Type checkers
     def isCaseClass(t: Type): Boolean = t.typeSymbol.isClass && t.typeSymbol.asClass.isCaseClass && !t.typeSymbol.isModuleClass
+    def isCaseObject(t: Type): Boolean = t.typeSymbol.isModuleClass && t.typeSymbol.asClass.isCaseClass && t.typeSymbol.isModuleClass
```

Comment:
```suggestion
    def isCaseObject(t: Type): Boolean = t.typeSymbol.isModuleClass && t.typeSymbol.asClass.isCaseClass
```
This is called twice

Reply:
Oops, fixed